src/EditorFeatures/CSharpTest/CodeActions/GenerateFromMembers/AddConstructorParameters/AddConstructorParametersTests.cs

```diff
@@ -67,5 +68,58 @@ protected override object CreateCodeRefactoringProvider(Workspace workspace)
+index: 0, parseOptions: TestOptions.Regular.WithTuplesFeature());
+        }
+
+        [Fact, Trait(Traits.Feature, Traits.Features.CodeActionsAddConstructorParameters), Trait(Traits.Feature, Test.Utilities.Traits.Features.Tuples)]
+        public async Task TestTupleWithNames()
+        {
+            await TestAsync(
+@"class Program { [|(int a, string b) i ; (string c, int d) s ;|] public Program ( (int a, string b) i ) { this . i = i ; } } ",
+@"class Program { (int a, string b) i ; (string c, int d) s ; public Program ( (int a, string b) i , (string c, int d) s ) { this . i = i ; this . s = s ; } } ",
+index: 0, parseOptions: TestOptions.Regular.WithTuplesFeature());
+        }
+
+        [Fact, Trait(Traits.Feature, Traits.Features.CodeActionsAddConstructorParameters), Trait(Traits.Feature, Test.Utilities.Traits.Features.Tuples)]
+        public async Task TestTupleWithDifferentNames()
+        {
+            await TestMissingAsync(
+@"class Program { [|(int a, string b) i ; (string c, int d) s ;|] public Program ( (int e, string f) i ) { this . i = i ; } } ",
+parseOptions: TestOptions.Regular.WithTuplesFeature());
+        }
+
+        [Fact, Trait(Traits.Feature, Traits.Features.CodeActionsAddConstructorParameters), Trait(Traits.Feature, Test.Utilities.Traits.Features.Tuples)]
+        public async Task TestTupleOptional()
+        {
+            await TestAsync(
+@"class Program { [|(int, string) i ; (string, int) s ;|] public Program ( (int, string) i ) { this . i = i ; } } ",
+@"class Program { (int, string) i ; (string, int) s ; public Program ( (int, string) i , (string, int) s = default((string, int)) ) { this . i = i ; this . s = s ; } } ",
+index: 1, parseOptions: TestOptions.Regular.WithTuplesFeature());
+        }
+
+        [Fact, Trait(Traits.Feature, Traits.Features.CodeActionsAddConstructorParameters), Trait(Traits.Feature, Test.Utilities.Traits.Features.Tuples)]
+        public async Task TestTupleOptionalWithNames()
+        {
+            // Note: default((string c, int d)) is less than ideal
+            await TestAsync(
+@"class Program { [|(int a, string b) i ; (string c, int d) s ;|] public Program ( (int a, string b) i ) { this . i = i ; } } ",
+@"class Program { (int a, string b) i ; (string c, int d) s ; public Program ( (int a, string b) i , (string c, int d) s = default((string c, int d)) ) { this . i = i ; this . s = s ; } } ",
+index: 1, parseOptions: TestOptions.Regular.WithTuplesFeature());
+        }
+
+        [Fact, Trait(Traits.Feature, Traits.Features.CodeActionsAddConstructorParameters), Trait(Traits.Feature, Test.Utilities.Traits.Features.Tuples)]
+        public async Task TestTupleOptionalWithDifferentNames()
+        {
+            await TestMissingAsync(
+@"class Program { [|(int a, string b) i ; (string c, int d) s ;|] public Program ( (int e, string f) i ) { this . i = i ; } } ",
+parseOptions: TestOptions.Regular.WithTuplesFeature());
+        }
```

Comment:
are nullable types and generic types legal in tuples? if so, they may make interesting tests.